crates/store/src/db/models/queries/notes.rs

```diff
@@ -0,0 +1,712 @@
+                &desired_block_num
+            ))
+            // filter the block's notes and return only the ones matching the requested tags or senders
+            .filter(
+                schema::notes::tag
+                .eq_any(&desired_note_tags)
+                .or(
+                    schema::notes::sender
+                    .eq_any(&desired_senders)
+                )
+            )
+            .get_results::<NoteSyncRecordRawRow>(conn)
+            .map_err(DatabaseError::from)?;
+
+    vec_raw_try_into(notes)
+}
+
+pub(crate) fn select_notes_by_id(
+    conn: &mut SqliteConnection,
+    note_ids: &[NoteId],
+) -> Result<Vec<NoteRecord>, DatabaseError> {
+    let note_ids = serialize_vec(note_ids);
+    // SELECT {}
+    // FROM notes
+    // LEFT JOIN note_scripts ON notes.script_root = note_scripts.script_root
+    // WHERE note_id IN rarray(?1),
+
+    let q = schema::notes::table
+        .left_join(
+            schema::note_scripts::table
+                .on(schema::notes::script_root.eq(schema::note_scripts::script_root.nullable())),
+        )
+        .filter(schema::notes::note_id.eq_any(&note_ids));
+    let raw: Vec<_> =
+        SelectDsl::select(q, (NoteRecordRaw::as_select(), schema::note_scripts::script.nullable()))
+            .load::<(NoteRecordRaw, Option<Vec<u8>>)>(conn)?;
+    let records = vec_raw_try_into::<NoteRecord, NoteRecordWithScriptRaw>(
+        raw.into_iter().map(NoteRecordWithScriptRaw::from),
+    )?;
+    Ok(records)
+}
+
+/// Select all notes from the DB using the given [`SqliteConnection`].
+///
+///
+/// # Returns
+///
+/// A vector with notes, or an error.
+#[cfg(test)]
+pub(crate) fn select_all_notes(
+    conn: &mut SqliteConnection,
+) -> Result<Vec<NoteRecord>, DatabaseError> {
+    // SELECT {cols}
+    // FROM notes
+    // LEFT JOIN note_scripts ON notes.script_root = note_scripts.script_root
+    // ORDER BY block_num ASC
+    let q = schema::notes::table.left_join(
+        schema::note_scripts::table
+            .on(schema::notes::script_root.eq(schema::note_scripts::script_root.nullable())),
+    );
+    let raw: Vec<_> =
+        SelectDsl::select(q, (NoteRecordRaw::as_select(), schema::note_scripts::script.nullable()))
+            .order(schema::notes::committed_at.asc())
+            .load::<(NoteRecordRaw, Option<Vec<u8>>)>(conn)?;
+    let records = vec_raw_try_into::<NoteRecord, NoteRecordWithScriptRaw>(
+        raw.into_iter().map(NoteRecordWithScriptRaw::from),
+    )?;
+    Ok(records)
+}
+
+/// Select note inclusion proofs matching the `NoteId`, using the given [`SqliteConnection`].
+///
+/// # Returns
+///
+/// - Empty map if no matching `note`.
+/// - Otherwise, note inclusion proofs, which `note_id` matches the `NoteId` as bytes.
+pub(crate) fn select_note_inclusion_proofs(
+    conn: &mut SqliteConnection,
+    note_ids: &BTreeSet<NoteId>,
+) -> Result<BTreeMap<NoteId, NoteInclusionProof>, DatabaseError> {
```

Comment:
We should mention in the doc comments that we impose a limit on the number of note IDs.

Also, we should probably start using `HashMap` and `HashSet` to improve performance.